lib/tunneler_tcp.c

```diff
@@ -82,11 +82,17 @@ static err_t on_tcp_client_data(void *io_ctx, struct tcp_pcb *pcb, struct pbuf *
         ZITI_LOG(INFO, "conn was closed err=%d", err);
         return ERR_OK;
     }
-    ZITI_LOG(DEBUG, "on_tcp_client_data status %d", err);
+    ZITI_LOG(VERBOSE, "status %d", err);
     struct io_ctx_s *_io_ctx = (struct io_ctx_s *)io_ctx;
     tunneler_io_context tnlr_io_ctx = *_io_ctx->tnlr_io_ctx_p;
 
+    if (tnlr_io_ctx == NULL) {
+        ZITI_LOG(INFO, "null tnlr_io_ctx");
```

Comment:
this isn't an ERROR?

Reply:
not necessarily. the connection could have been closed from the "ziti" side of the tunnel, which would null out the tunneler i/o context.